homeassistant/components/madvr/binary_sensor.py

```diff
@@ -0,0 +1,144 @@
+"""Binary sensor entities for the madVR integration."""
+
+from homeassistant.components.binary_sensor import BinarySensorEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import MadVRCoordinator
+
+type MadVRConfigEntry = ConfigEntry[MadVRCoordinator]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: MadVRConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the binary sensor entities."""
+    coordinator = entry.runtime_data
+    async_add_entities(
+        [
+            MadvrPowerStateBinarySensor(coordinator),
+            MadvrSignalStateBinarySensor(coordinator),
+            MadvrHDRFlagBinarySensor(coordinator),
+            MadvrHDROutgoingFlagBinarySensor(coordinator),
```

Comment:
Use entity descriptions. That means you have 1 generic sensor entity, and then change what they return based on the entity description injected.

You can also reap a lot of beneifts of creating a base entity, which is just the CoordinatorEntity, which you can use as base entity for all your other entities, so you don't have to set `device_info` and `has_entity_name` everywhere